libtransact/src/context/mod.rs

```diff
@@ -22,23 +22,27 @@ pub type ContextId = [u8; 16];
 mod error;
 pub mod manager;
 
-use crate::context::manager::ContextManagerError;
+use crate::context::manager::thread::ContextManagerThreadError;
 use crate::protocol::receipt::{Event, StateChange, TransactionReceipt};
 use std::mem;
 use uuid::Uuid;
 
 /// ContextManager functionality used by the Scheduler.
 pub trait ContextLifecycle: Send {
     /// Create a new Context, returning a unique ContextId.
-    fn create_context(&mut self, dependent_contexts: &[ContextId], state_id: &str) -> ContextId;
+    fn create_context(
+        &self,
+        dependent_contexts: &[ContextId],
+        state_id: &str,
+    ) -> Result<ContextId, ContextManagerThreadError>;
 
-    fn drop_context(&mut self, context_id: ContextId);
+    fn drop_context(&self, context_id: ContextId);
 
     fn get_transaction_receipt(
         &self,
         context_id: &ContextId,
         transaction_id: &str,
-    ) -> Result<TransactionReceipt, ContextManagerError>;
+    ) -> Result<TransactionReceipt, ContextManagerThreadError>;
```

Comment:
These should remain `ContextManagerError`, from the perspective of the trait caller.  In other words, the user of the trait shouldn't need to know anything about the underlying implementation.